test/test_linalg.py

```diff
@@ -3991,7 +3991,8 @@ def _gen_shape_inputs_linalg_triangular_solve(self, shape, dtype, device, well_c
                 size_b = size_b[1:]
 
             if well_conditioned:
-                PLU = torch.linalg.lu(make_randn(*size_a))
+                U, _, Vh = torch.linalg.svd(make_randn(*size_a), full_matrices=False)
+                PLU = torch.linalg.lu(U @ Vh)
```

Comment:
Probably better to simply use
```python
make_fullrank = make_fullrank_matrices_with_distinct_singular_values
```
You can set this outside the loop and use it here.

Reply:
The goal is to have condition number 1, not just full-rank, as it is not sufficient for these tests...

Reply:
Not quite, we just want something that can be solved. Note that we're taking just below the `U` part if `not uni`, so that will almost never have a condition number of 1.